src/databricks/labs/ucx/mixins/fixtures.py

```diff
@@ -609,11 +609,13 @@ def create(*, url=None, provider=None, path=None, **kwargs):
 
 @pytest.fixture
 def make_user(ws, make_random):
-    yield from factory(
-        "workspace user",
-        lambda **kwargs: ws.users.create(user_name=f"sdk-{make_random(4)}@example.com".lower(), **kwargs),
-        lambda item: ws.users.delete(item.id),
-    )
+
+    @retried(on=[ResourceConflict], timeout=timedelta(seconds=30))
+    def create(**kwargs):
+        user = ws.users.create(user_name=f"sdk-{make_random(4)}@example.com".lower(), **kwargs)
```

Comment:
we could make it 8 to reduce collision chance
```suggestion
        user = ws.users.create(user_name=f"sdk-{make_random(8)}@example.com".lower(), **kwargs)
```